### ISSUE TITLE
brave://rewards ads per hour setting should not change to 0 after upgrade

### ISSUE THREAD

**Issue by tmancey** [brave/brave-browser#17155]
## Description 

brave://rewards ads per hour setting should not change to 0 after upgrade

## Steps to Reproduce

   1. Fresh install of 1.20.x or 1.21.x
   2. Join Rewards/Ads
   3. Change ads per hour anything other than 2
   4. Change ads per hour to 2
   5. Upgrade to a version of the browser between 1.22.x and 1.26.x
   6. Upgrade to 1.27.x (release), 1.28.x (beta) or 1.29.x (nightly)

## Actual result:

Ads per hour are changed to 0

## Expected result:
Ads per hour should not change

## Reproduces how often: 

Easily reproduced

## Version/Channel Information:

- Can you reproduce this issue with the current release? Yes
- Can you reproduce this issue with the beta channel? Yes
- Can you reproduce this issue with the nightly channel? Yes

## Other Additional Information:

- Does the issue resolve itself when disabling Brave Shields? N/A
- Does the issue resolve itself when disabling Brave Rewards? N/A
- Is the issue reproducible on the latest version of Chrome? N/A

**Comment by GeetaSarvadnya** [comment]
Verification passed on

```

Brave | 1.27.109 Chromium: 92.0.4515.115 (Official Build) (64-bit)
-- | --
Revision | 48cb2f4029b84b003719740a6cf9ca73f374a857-refs/branch-heads/4515_105@{#4}
OS | Windows 10 OS Version 2004 (Build 19041.1110)


```
- Reproduced the issue in `1.27.108`
 
1. Enabled rewards in `1.21.76` and changed defaults ads per hours settings `2` to `3` and then set it back to `2`
2. `Ads per hour` settings changed to `5` when `1.21.76` is upgraded to `1.25.73` 
3. `Ads per hour` settings changed to `0` when `1.25.73` is upgraded to `1.27.108` 

- Confirmed Issue is fixed and working fine in `1.27.109`
1. Enabled rewards in `1.21.76` and changed defaults ads per hours settings `2` to `3` and then set it back to `2`
2. `Ads per hour` settings changed to `5` when `1.21.76` is upgraded to `1.25.73` 
3. `Ads per hour` settings remained `5` when `1.25.73` is upgraded to `1.27.108`

**Comment by srirambv** [comment]
Verification passed on Oppo R5 with Android 11 running 1.27.109 x64 build

### `Test Case #1 (Original reported issue)`
- Installed `1.21.76 Chromium: 89.0.4389.86` and enabled rewards, ads is set to 2 ads per hour as default
- Changed Ads to 5 ads per hour
- Changed Ads to 2 ads per hour
- Upgraded to `1.25.73 Chromium: 91.0.4472.106`
- Checked Ads settings is set to 5 ads per hour
- Upgraded to `1.27.108 Chromium: 92.0.4515.107`
- Checked Ads settings is reset to 0
- Upgraded to `1.27.109 Chromium: 92.0.4514.115`
- Checked Ads settings is set back to 5 ads per hour

https://user-images.githubusercontent.com/17010094/127138636-3e3029c1-b223-4dcf-a76e-2730457cb243.mp4

### `Test Case #2 (ads per hour set as 0 before upgrading)`
- Install `1.27.108 Chromium: 92.0.4515.107` and set ads per hour to 0
- Upgraded to `1.29.31 Chromium: 92.0.4515.115` and ensure that ads per hour was changed to 5

###  `Test Case 3 (ads per hour set as 5 before upgrading - Default)`
- Install `1.27.108 Chromium: 92.0.4515.107` and leave ads per hour as 5 which is the default
- Upgraded  to `1.29.31 Chromium: 92.0.4515.115` and ensure that ads per hour is still set as 5

### `Test Case 4 (ads per hour set as 1 before upgrading - Custom Selection)`
- Install `1.27.108 Chromium: 92.0.4515.107` and set ads per hour to 1
- Upgraded to `1.29.31 Chromium: 92.0.4515.115 `and ensure that ads per hour is still set as 1

### `Test Case 5 (ads per hour set as 10 before upgrading - Custom Selection)`
- Install `1.27.108 Chromium: 92.0.4515.107` and set ads per hour to 10
- Upgraded  to `1.29.31 Chromium: 92.0.4515.115` and ensure that ads per hour is still set as 10

**Verified the above scenarios by setting ads per hour via rewards onboarding and manually changing the settings**

-------------------

Verification passed on Samsung Tab A with Android 11 running 1.27.109 x64 build

### `Test Case #1 (Original reported issue)`
- Installed `1.21.76 Chromium: 89.0.4389.86` and enabled rewards, ads is set to 2 ads per hour as default
- Changed Ads to 5 ads per hour
- Changed Ads to 2 ads per hour
- Upgraded to `1.25.72 Chromium: 91.0.4472.106`
- Checked Ads settings is set to 5 ads per hour
- Upgraded to `1.27.108 Chromium: 92.0.4515.107`
- Checked Ads settings is reset to 0
- Upgraded to `1.27.109 Chromium: 92.0.4514.115`
- Checked Ads settings is set back to 5 ads per hour

https://user-images.githubusercontent.com/17010094/127146842-154725c2-cee5-4ccb-9c8a-85cbc12cf290.mp4


### `Test Case #2 (ads per hour set as 0 before upgrading)`
- Install `1.27.108 Chromium: 92.0.4515.107` and set ads per hour to 0
- Upgraded to `1.29.31 Chromium: 92.0.4515.115` and ensure that ads per hour was changed to 5

###  `Test Case 3 (ads per hour set as 5 before upgrading - Default)`
- Install `1.27.108 Chromium: 92.0.4515.107` and leave ads per hour as 5 which is the default
- Upgraded  to `1.29.31 Chromium: 92.0.4515.115` and ensure that ads per hour is still set as 5

### `Test Case 4 (ads per hour set as 1 before upgrading - Custom Selection)`
- Install `1.27.108 Chromium: 92.0.4515.107` and set ads per hour to 1
- Upgraded to `1.29.31 Chromium: 92.0.4515.115 `and ensure that ads per hour is still set as 1

### `Test Case 5 (ads per hour set as 10 before upgrading - Custom Selection)`
- Install `1.27.108 Chromium: 92.0.4515.107` and set ads per hour to 10
- Upgraded  to `1.29.31 Chromium: 92.0.4515.115` and ensure that ads per hour is still set as 10

**Verified the above scenarios by setting ads per hour via rewards onboarding and manually changing the settings**
